packages/elastic_agent/data_stream/apm_server_logs/manifest.yml

```
@@ -5,3 +5,5 @@ elasticsearch:
   index_template:
     mappings:
       dynamic: false
+      _source:
+        mode: synthetic
```

I think this syntax is not valid in recent versions of the spec. Please use the following syntax, Fleet is aware of the `source_mode` setting and can handle it in a safer way.
```suggestion
  source_mode: synthetic
```

---

Interesting, it worked well on all my manual testing. But I'll update and test again.

---

Yes, this enables synthetic mode, but Fleet may not be aware of it. There is some logic in Fleet to allow to override this option by the user, not sure if this works with this syntax.
This behavior is described here: https://github.com/elastic/kibana/issues/141211#issue-1380774605

Older versions of the spec, like 1.0.0, were much less restrictive, and allowed things that could lead to problematic or ambiguous situations. In v2 and v3 we have been introducing restrictions to avoid these situations. For example restrict now what can be configured for index templates in the manifest.

You can find here some help for some common issues when upgrading packages to v2 or v3: https://github.com/elastic/elastic-package/blob/main/docs/howto/update_major_package_spec.md